src/VitessQueryValidator.php

```diff
@@ -37,20 +38,55 @@ public static function validate(Query $query, AsyncMysqlConnection $conn): void
         if ($query is SelectQuery) {
             /*HHAST_FIXME[DontUseAsioJoin]*/
             \HH\Asio\join((new SelectQueryValidator($query, $conn))->processHandlers());
-        }
+		} else if ($query is UpdateQuery) {
+            \HH\Asio\join((new UpdateQueryValidator($query, $conn))->processHandlers());
+		}
     }
 
     public static function extractColumnExprNames(vec<Expression> $selectExpressions): keyset<string> {
         $exprNames = keyset[];
         foreach ($selectExpressions as $expr) {
             if ($expr is ColumnExpression) {
                 $exprNames[] = $expr->name;
-            }
+			} elseif ($expr is BinaryOperatorExpression) {
+				$exprNames[] = $expr->left->name;
+		   	}
         }
         return $exprNames;
     }
 }
 
+final class UpdateQueryValidator extends VitessQueryValidator {
+    public function __construct(public UpdateQuery $query, public AsyncMysqlConnection $conn) {}
+
+    <<__Override>>
+    public function getHandlers(): dict<string, (function(): Awaitable<void>)> {
+		return dict[
+			UnsupportedCases::PRIMARY_VINDEX_COLUMN => inst_meth($this, 'updateChangesPrimaryVindexColumn')
+		];
+	}
+
+	public async function updateChangesPrimaryVindexColumn(): Awaitable<void> {
+    	$set = $this->query->setClause;
+
+    	list($database, $table_name) = Query::parseTableName($this->conn, $this->query->updateClause['name']);
+        $table_schema = QueryContext::getSchema($database, $table_name);
+        $vitess_sharding = $table_schema['vitess_sharding'] ?? null;
+
+		if ($vitess_sharding === null) {
+			throw new SQLFakeVitessQueryViolation(Str\format("Missing Vitess sharding information for: %s", $table_name));
```

Comment:
do we need to throw here? should we just silently allow it if the sharding information isn't available, or at least only complain if strict schema mode is enabled?